### ISSUE TITLE
Add support for reexport operator suggestions

### ISSUE THREAD

**Issue by ChShersh** [ndmitchell/hlint#526]
I'm working on alternative `Prelude` where I reexport some useful things. And I would like to provide suggestions for users to not export some functions and operators from modules explicitly when they are using my prelude.

Consider the following `.hlint-test.yaml` file:

```yaml
- warn: { name: "Use '>>>' from Relude"
        , lhs: (Control.Category.>>>)
        , rhs: (>>>)
        }
```

When I have the following `Test.hs` file:

```haskell
import Control.Category ((>>>))
import Relude

foo = (>>>)
```

everything works as expected and I see expected hint message:

```
Test.hs:4:7: Warning: Use '>>>' from Relude
Found:
  (>>>)
Perhaps:
  (>>>)
```

But when I'm applying operator to arguments:

```haskell
import Control.Category ((>>>))
import Relude

foo = id >>> id
```

I see only `No hints`. Though, it would be really great if I can observe the hint in the second case as well!

**Comment by ndmitchell** [comment]
I can see why that's useful, should it just be a special case in match? Should it also match `(>>> a)` and `(>>> a)`? Does `map` work in backticks to match with a `LHS=map` rule?

**Comment by ChShersh** [comment]
> Should it also match (a >>>) and (>>> a)?

In my case I would to match all of this is well. But I guess that something smarter is required in general case, when you want to replace weird operator with more meaningful name.

```yaml
- warn: { lhs: (~@@^>), rhs: queryFoo }
```

> Does map work in backticks to match with a LHS=map rule?

Unfortunately, no 😞 If I have rule like this:

```yaml
- warn: { lhs: map, rhs: foo }
```

Then the following code triggers HLint warning:

```haskell
bar = map even
```

But this one doesn't not:

```haskell
bar = (even `map`)
```

**Comment by ndmitchell** [comment]
It sounds like this is turning into a name replacement hint. Should that be something? Should it have different semantics to normal LHS/RHS hints? Should they be tagged specially? It's not immediately clear to me.
